var/spack/repos/builtin/packages/dd4hep/package.py

```diff
@@ -50,9 +51,20 @@ class Dd4hep(CMakePackage):
     # See https://github.com/spack/spack/issues/24232
     patch('cmake_language.patch', when='@:1.17')
 
+    # variants for subpackages
+    variant('ddcad', default=True, description="Enable CAD interface based on Assimp")
+    variant('ddg4', default=True, description="Enable the simulation part based on Geant4")
+    variant('ddg4', default=True, description="Enable the simulation part based on Geant4")
+    variant('ddrec', default=True, description="Build DDRec subpackage.")
+    variant('dddetectors', default=True, description="Build DDDetectors subpackage.")
+    variant('ddcond', default=True, description="Build DDCond subpackage.")
+    variant('ddalign', default=True, description="Build DDAlign subpackage.")
+    variant('dddigi', default=True, description="Build DDDigi subpackage.")
+    variant('ddeve', default=True, description="Build DDEve subpackage.")
+    variant('utilityapps', default=True, description='Build UtilityApps subpackage.')
```

Comment:
Are these variants, and the new `subpackages` build argument later, supported on all versions of DD4Hep that the spack package lists?

Reply:
@vvolkl :point_up:

Reply:
That's a very good point. DDCAD was only added in  v0-12-01, so the older ones definitely won't work (The option itself was added in v0-11-01, but will most likely just be ignored for older versions). I did a few test builds, and the old versions actually do not build for unrelated reasons (probably the ROOT version) so it will be good to update the dependencies for those I want to keep, I'll give that a go.

I guess there is no need to keep the very old versions though, so if you agree I'd drop everything older than v0-12-01.

Reply:
As long as you're not aware of any clients relying on versions before v0-12-01, I think that would be fine (@alalazo, do we have a formal policy for removing "obsolete" or otherwise no longer supported versions?). Sounds like they can't be built anyway, so perhaps thats moot.

Reply:
No, I don't think anyone uses those versions anymore.I think one can add `deprecated=True` to individual versions, and remove them in the next release, which might be the best thing here?

Reply:
Adding `deprecated` with removal in the next release sounds good to me, so if you want to add that to the affected versions, I think this will be good to go!